xdsl/printer.py

```diff
@@ -296,7 +297,7 @@ def print_paramattr_parameters(
         self.print(">")
 
     def print_string_literal(self, string: str):
-        self.print(f'"{string}"')
+        self.print(json.dumps(string))
```

Comment:
I would argue that the string should be already escaped at that point (if that is the reason you use `json` here).
So I would just keep what we had before, and make sure the string is escaped by the parser.

Reply:
If you have a better library function we could use, please make a suggestion!

Reply:
Something like `print(string.encode('unicode_escape'))` should probably do the trick!

Reply:
This does not escape `"`, and so it fails :(